### PR TITLE
Prohibit whitespace-only names

### LINKED ISSUE #236: Prohibit whitespace-only names
Those are not rendered to HTML, so it messes with the markup at places. Generally not a good idea to allow such things.

### PR DESCRIPTION
**Pull request by illright**
Closes #236 
Will also rename the Moderation internal activity to `[[Moderation]]`, because I figured that in some cases people would wanna create the Moderation activity for themselves, so to avoid collisions I added symbols.

### CODE REVIEW

File: src/components/projects/new/edit-activity.svelte [hunk 1]
```diff
@@ -63,11 +63,11 @@
   }
 
   function validateAndSave() {
-    if (!activity.name) {
+    if (!activity.name || /^\s*$/.test(activity.name)) {
```
Comment (aabounegm) [inline src/components/projects/new/edit-activity.svelte:66]: Regex is slower than native methods
```suggestion
    if (!activity.name.trim()) {
```

File: src/components/projects/new/edit-activity.svelte [hunk 2]
```diff
@@ -118,7 +118,7 @@
       title="Activity name"
       required
       error={
-        (activity.name === '' && "The name must not be empty.")
+        (/^\s*$/.test(activity.name) && "The name must not be empty.")
```
Comment (aabounegm) [inline src/components/projects/new/edit-activity.svelte:121]: Same here
```suggestion
        (!activity.name.trim() && "The name must not be empty.")
```

File: src/containers/projects/new/step-1.svelte [hunk 3]
```diff
@@ -25,7 +25,7 @@
     required
     error={
          (duplicateName && "The name must be unique.")
-      || ($project.name === '' && "The name must not be empty.")
+      || (/^\s*$/.test($project.name) && "The name must not be empty.")
```
Comment (aabounegm) [inline src/containers/projects/new/step-1.svelte:28]:
```suggestion
      || (!$project.name.trim() && "The name must not be empty.")
```

File: src/routes/products/[id([0-9]+)]/edit.svelte [hunk 4]
```diff
@@ -190,14 +190,14 @@
   }
 
   async function saveChanges() {
-    if (!product.name) {
+    if (!product.name || /^\s*$/.test(product.name)) {
```
Comment (aabounegm) [inline src/routes/products/[id([0-9]+)]/edit.svelte:193]:
```suggestion
    if (!product.name.trim()) {
```

File: src/routes/products/new.svelte [hunk 5]
```diff
@@ -135,14 +135,14 @@
   async function createProduct() {
     errorMessage = null;
     prefetch('/store');
-    if (!product.name) {
+    if (!product.name || /^\s*$/.test(product.name)) {
```
Comment (aabounegm) [inline src/routes/products/new.svelte:138]:
```suggestion
    if (!product.name.trim()) {
```

File: src/routes/projects/[id([0-9]+)]/edit.svelte [hunk 6]
```diff
@@ -143,7 +143,7 @@
         required
         error={
              (duplicateName && "The name must be unique.")
-          || (project.name === '' && "The name must not be empty.")
+          || (/^\s*$/.test(project.name) && "The name must not be empty.")
```
Comment (aabounegm) [inline src/routes/projects/[id([0-9]+)]/edit.svelte:146]:
```suggestion
          || (!project.name.trim() && "The name must not be empty.")
```

File: src/routes/products/[id([0-9]+)]/edit.svelte (hunk 4 above)
Comment (aabounegm) [reply src/routes/products/[id([0-9]+)]/edit.svelte:193]: Also, if `product.name` can be `null`, the RegEx test can deal with it. No need to check it first

Comment (illright) [reply src/routes/products/[id([0-9]+)]/edit.svelte:193]: But it will say `false`, while I want it to say `true` in that "true, this is a bad name"

**Pull request merged**